paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowControllerInitializer.kt

```diff
@@ -92,99 +84,89 @@ internal class DefaultFlowControllerInitializer @Inject constructor(
         }?.isReady()?.first() ?: false
     }
 
-    private suspend fun createWithCustomer(
+    private suspend fun create(
         clientSecret: ClientSecret,
         stripeIntent: StripeIntent,
-        customerConfig: PaymentSheet.CustomerConfiguration,
+        customerConfig: PaymentSheet.CustomerConfiguration?,
         config: PaymentSheet.Configuration?,
         isGooglePayReady: Boolean,
         isLinkReady: Boolean
     ): FlowControllerInitializer.InitResult {
         val prefsRepository = prefsRepositoryFactory(customerConfig)
 
-        val paymentMethodTypes = getSupportedSavedCustomerPMs(
-            stripeIntent,
-            config,
-            lpmResourceRepository.getRepository()
-        ).mapNotNull {
-            // The SDK is only able to parse customer LPMs
-            // that are hard coded in the SDK.
-            PaymentMethod.Type.fromCode(it.code)
+        val paymentMethods = if (customerConfig != null) {
+            retrieveCustomerPaymentMethods(
+                stripeIntent,
+                config,
+                customerConfig
+            )
+        } else {
+            emptyList()
         }
 
-        return customerRepository.getPaymentMethods(
-            customerConfig,
-            paymentMethodTypes
-        ).filter { paymentMethod ->
-            paymentMethod.hasExpectedDetails()
-        }.let { paymentMethods ->
+        val savedSelection = prefsRepository.getSavedSelection(isGooglePayReady, isLinkReady)
+        if (savedSelection == SavedSelection.None) {
+            // No saved selection has been set yet, so we'll initialize it with a default
+            // value based on which payment methods are available.
             setLastSavedPaymentMethod(
                 prefsRepository,
                 isGooglePayReady,
                 isLinkReady,
                 paymentMethods
             )
-
-            FlowControllerInitializer.InitResult.Success(
-                InitData(
-                    config = config,
-                    clientSecret = clientSecret,
-                    stripeIntent = stripeIntent,
-                    paymentMethods = paymentMethods,
-                    savedSelection = prefsRepository.getSavedSelection(
-                        isGooglePayReady,
-                        isLinkReady
-                    ),
-                    isGooglePayReady = isGooglePayReady
-                )
-            )
-        }
-    }
-
-    private fun createWithoutCustomer(
-        clientSecret: ClientSecret,
-        stripeIntent: StripeIntent,
-        config: PaymentSheet.Configuration?,
-        isGooglePayReady: Boolean,
-        isLinkReady: Boolean
-    ): FlowControllerInitializer.InitResult {
-        val savedSelection = if (isLinkReady) {
-            SavedSelection.Link
-        } else if (isGooglePayReady) {
-            SavedSelection.GooglePay
-        } else {
-            SavedSelection.None
         }
```

Comment:
Removed this logic for guests, because we can just use `setLastSavedPaymentMethod()` for this.